### PR TITLE
WIP: Add more consumer session metrics

### PR DESCRIPTION
**Pull request by anjmao**
Part of https://github.com/mysteriumnetwork/node/issues/1298

This PR implements logic to send session connectivity status metrics to quality oracle.

### CODE REVIEW

File: core/connection/event.go [hunk 1]
```diff
@@ -17,31 +17,44 @@
...
-	// SessionCreatedStatus represents a session creation event
-	SessionCreatedStatus = "Created"
-	// SessionEndedStatus represents a session end
-	SessionEndedStatus = "Ended"
+	// SessionStatusCreated represents a session creation event.
+	SessionStatusCreated = SessionStatus("Created")
+	// SessionStatusEnded represents a session end.
+	SessionStatusEnded = SessionStatus("Ended")
+	// SessionStatusIpNotChanged represents what ip is not changed after session created.
+	SessionStatusIpNotChanged = SessionStatus("IpNotChanged")
```
Comment (soffokl) [inline core/connection/event.go]:
```suggestion
	SessionStatusIPNotChanged = SessionStatus("IpNotChanged")
```

File: consumer/bandwidth/tracker.go [hunk 2]
```diff
@@ -94,7 +94,7 @@ func (t *Tracker) ConsumeSessionEvent(sessionEvent connection.SessionEvent) {
 	t.lock.Lock()
 	defer t.lock.Unlock()
 	switch sessionEvent.Status {
-	case connection.SessionEndedStatus, connection.SessionCreatedStatus:
+	case connection.SessionStatusEnded, connection.SessionStatusCreated:
```
[the remark this answers is not in the document]
Comment (anjmao) [reply consumer/bandwidth/tracker.go]: Sorry, you are right. Next time I will to such refactoring in a separate PR. 

Regarding naming there are some common naming patters in go especially for go const which you will find in 90% of open source projects. In node code base in some places naming is not consistent eg. [here](https://github.com/mysteriumnetwork/node/blob/master/core/connection/manager.go#L38) we prefix with Err which is good.

Consider this example there you have public const values for possible event topics.
```go
package connection
const (
	StateEventTopic = "State"
	StatisticsEventTopic = "Statistics"
	SessionEventTopic = "Session"
)
```
Now if you use it in another package you can't instantly get all possible topics from code completion

```go
connection.StateEventTopic
``` 

If we prefix with semantic meaning:
```go
const (
	EventTopicState = "State"
	EventTopicStatistics = "Statistics"
	EventTopicSession = "Session"
)
```
We can get all event topics by typing connection.EventTopic*
```go
connection.EventTopic*
```

[the remark this answers is not in the document]
Comment (zolia) [reply consumer/bandwidth/tracker.go]: agree, that is why i suggest too that semantic type of constant should go first as a prefix.

**Comment by anjmao** [comment]
@zolia It's a good point. I wasn't able to get problematic case probably because I was testing on docker network locally but it can happen for sure. As ip check is already implemented in a blocking way and merged to master I suggest we merge this PR and I do other changes in separate PR and this time try to not touch irrelevant code with refactoring stuff :)

**Comment by tadaskay** [comment]
@zolia It's fine to do refactoring as long as it's done in separate commits. This behavior should be encouraged because in your proposal, it usually never comes to the second PR (refactoring) and as a result, no refactoring is done. 

You can review commit-by-commit to reduce the scope of the files under review at a time.

**Pull request closed without merging**